networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/gossip/forks/GossipForkManager.java

```diff
@@ -0,0 +1,223 @@
+            () ->
+                LOG.warn(
+                    "Not publishing block because no gossip subscriptions are active for slot {}",
+                    block.getSlot()));
+  }
+
+  public synchronized void subscribeToAttestationSubnetId(final int subnetId) {
+    if (currentSubnetSubscriptions.add(subnetId)) {
+      activeSubscriptions.forEach(
+          subscription -> subscription.subscribeToAttestationSubnetId(subnetId));
+    }
+  }
+
+  public void unsubscribeFromAttestationSubnetId(final int subnetId) {
+    if (currentSubnetSubscriptions.remove(subnetId)) {
+      activeSubscriptions.forEach(
+          subscription -> subscription.unsubscribeFromAttestationSubnetId(subnetId));
+    }
+  }
+
+  private boolean isActive(final GossipForkSubscriptions subscriptions) {
+    return activeSubscriptions.contains(subscriptions);
+  }
+
+  private void startSubscriptions(final GossipForkSubscriptions subscription) {
+    if (activeSubscriptions.add(subscription)) {
+      subscription.startGossip(
+          recentChainData.getGenesisData().orElseThrow().getGenesisValidatorsRoot());
+      currentSubnetSubscriptions.forEach(subscription::subscribeToAttestationSubnetId);
+    }
+  }
+
+  private void stopSubscriptions(final GossipForkSubscriptions subscriptions) {
+    if (activeSubscriptions.remove(subscriptions)) {
+      subscriptions.stopGossip();
+    }
+  }
+
+  private Optional<GossipForkSubscriptions> getSubscriptionActiveAtSlot(final UInt64 slot) {
+    final UInt64 epoch = spec.computeEpochAtSlot(slot);
+    return getSubscriptionActiveAtEpoch(epoch);
+  }
+
+  private Optional<GossipForkSubscriptions> getSubscriptionActiveAtEpoch(final UInt64 epoch) {
+    return Optional.ofNullable(forksByActivationEpoch.floorEntry(epoch)).map(Map.Entry::getValue);
+  }
+
+  public static class Builder {
+
+    private Spec spec;
+    private RecentChainData recentChainData;
+    private final NavigableMap<UInt64, GossipForkSubscriptions> forksByActivationEpoch =
+        new TreeMap<>();
+
+    public Builder spec(final Spec spec) {
+      this.spec = spec;
+      return this;
+    }
+
+    public Builder recentChainData(final RecentChainData recentChainData) {
+      this.recentChainData = recentChainData;
+      return this;
+    }
+
+    public Builder fork(final GossipForkSubscriptions forkSubscriptions) {
+      final UInt64 activationEpoch = forkSubscriptions.getActivationEpoch();
+      checkState(
+          !forksByActivationEpoch.containsKey(activationEpoch),
+          "Can not schedule two forks to activate at the same epoch");
+      forksByActivationEpoch.put(activationEpoch, forkSubscriptions);
+      return this;
+    }
+
+    public GossipForkManager build() {
+      checkNotNull(spec, "Must supply spec");
+      checkNotNull(recentChainData, "Must supply recentChainData");
+      checkState(!forksByActivationEpoch.isEmpty(), "Must specify at least one fork");
```

Comment:
Suggest we stream the forks from `ForkManifest` and validate that we have a set of subscriptions for each scheduled fork.